Sources/Overload/OvEditor/src/OvEditor/Core/EditorActions.cpp

```diff
@@ -970,8 +972,8 @@ void OvEditor::Core::EditorActions::PropagateFileRenameThroughSavedFilesOfType(c
 				out.close(); in.close();
 			}
 
-			std::filesystem::copy_file("TEMP", entry.path(), std::filesystem::copy_options::overwrite_existing);
-			std::filesystem::remove("TEMP");
+			std::filesystem::copy_file(tempFilePath.string(), entry.path(), std::filesystem::copy_options::overwrite_existing);
+			std::filesystem::remove(tempFilePath.string());
```

Comment:
I think we could use `c_str()` instead of `string()` whenever possible to prevent multiple allocation.
We're limited by `PathParser::GetFileType()` and `OVLOG_INFO`  which expects `std::string` down the line.

Calling `.string()` will call `_Convert_wide_to` which then call `_Convert_wide_to_narrow` for the char underlying type, which makes allocation. Unless we need the features embedded in the std::string type, we should try to call `c_str()` or at very least get the real type in `std::filesystem`, giving a const reference to _Text which is a `std::wstring`, by calling `native()` instead of `string()`.

Same applies in `ifstream in` and `ofstream out` above.

Reply:
Updated the implementation, removed unnecessary `string()` calls and also removed the need for a temp file.